resources/utils/fetch.ts

```diff
@@ -0,0 +1,7 @@
+import fetch, { RequestInit } from 'node-fetch'
+
+export async function fetchWithTimeout (url: string, options: RequestInit, timeout: number) {  
+  const controller = new AbortController() 
+  setTimeout(() => controller.abort(), timeout)
+  return fetch(url, { ...options, signal: controller.signal as RequestInit['signal'] })
```

Comment:
can we just use the `AbortSignal` type directly instead of referencing a type from another interface?

Reply:
Nope, doesn't work - but feel free to give it a go yourself.

Reply:
done, what did you mean by "it doesn't work"?

Reply:
I get a TS error:

```
Type 'AbortSignal' is not assignable to type 'import("/Users/sam/workspace/frame3/node_modules/@types/node-fetch/externals").AbortSignal'.
  Types of property 'onabort' are incompatible.
    Type '((this: AbortSignal, ev: Event) => any) | null' is not assignable to type '((this: AbortSignal, event: any) => void) | null'.
      Type '(this: AbortSignal, ev: Event) => any' is not assignable to type '(this: AbortSignal, event: any) => void'.
        The 'this' types of each signature are incompatible.
          Type 'AbortSignal' is missing the following properties from type 'AbortSignal': reason, throwIfAbortedts(2322)
```

Reply:
maybe you were using the global `AbortSignal` type and not the same one you were using before which is the node-fetch version

Reply:
ok, nice fix